<a href="https://colab.research.google.com/github/ju4n1t0x/agroindustria/blob/main/CDD_Grupo24_TPIntegrador.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Trabajo Práctico Integrador - Ciencia de Datos

**Nombre y Apellido:** Stefania Martos, Juan Ignacio Sasia, Juan José Muñoz Franchi, Jorge Nicolas Segovia. <br>
**Materia:** Ciencia de Datos  <br>
**Docente:** Germán Giró, Malena Constancio, Giselle Bustamante<br>
**Fecha:** 26/04/2026  <br>
**Unidad:** Módulo 1 y 2  <br>
**Número de ejercicio/actividad:** Entrega 1  

## 1. Objetivo

El objetivo de este trabajo es analizar un conjunto de datos reales sobre movimientos de ganado bovino en Argentina, con la intención de entender cómo funciona este sistema y qué tipo de patrones pueden aparecer.

En esta primera entrega, el foco está en definir bien el problema, entender el contexto del dominio y hacer una primera exploración del dataset para ver qué tipo de información contiene y en qué estado se encuentra.

## 2. Dataset

Fuente:  
- Año 2018:
https://datos.gob.ar/dataset/agroindustria-senasa---movimientos-bovinos/archivo/agroindustria_81330b51-8dd9-420d-842e-aefe56151d3a
- Año 2017:
https://datos.gob.ar/dataset/agroindustria-senasa---movimientos-bovinos/archivo/agroindustria_9cd595bd-c2f4-46a5-993f-3b84f75f1b3f
- Año 2016:
https://datos.gob.ar/dataset/agroindustria-senasa---movimientos-bovinos/archivo/agroindustria_9e1e089c-668d-4614-8289-40aaba1efbfc

##3.1.A - Comprensión del negocio


###Contexto / dominio del problema

El presente trabajo pertenece al area de la produccion y logistica ganadera bovina en Argentina, un punto clave de la agroindustria nacional.

El dataset proviene de SENASA(Servicio Nacional de Sanidad y Calidad Agroalimentaria) y contiene informacion sobre movimientos mensuales de ganado bovino entre distintas regiones del pais, desagregador por categoria animal (vacas, novillos, terneros, etc).

Estos movimientos corresponden a distintas etapas del ciclo productivo como: **cria, recria, engorde (por ejemplo en feedlots) y traslado a faena.**

El seguimiento de estos movimientos es fundamental para:

- el control sanitario  
- la organización logistica del sector  
- la sostenibilidad de la cadena productiva  
- entre otros  

Diversos actores pueden beneficiarse de este analisis:

- organismos publicos (control y regulacion)  
- productores ganaderos (toma de decisiones)  
- frigorificos (planificacion de abastecimiento)  
- investigadores (modelado de redes y riesgos sanitarios)  


###Problema o pregunta que se busca responder con los datos:

El sistema de movimientos bovinos en Argentina es parte de una red compleja que involucra diferentes regiones y etapas productivas.

Sin embargo, no se entiende claramente cómo está organizada esta red ni cómo evoluciona en el tiempo, lo que dificulta tanto su análisis como la toma de decisiones en situaciones críticas.

Esta falta de comprensión impacta la sanidad animal, la eficiencia logística y el funcionamiento del sistema productivo y comercial del sector agrónomo.



###Objetivo general

El objetivo general de nuestro trabajo es analizar la estructura y la dinámica de los movimientos bovinos en Argentina para identificar patrones relevantes, que contribuyan a comprender el sistema logístico y productivo del sector ganadero.


###Objetivos específicos

- Identificar las regiones con mayor participación en los flujos de ganado, destacando posibles nodos relevantes dentro de la red.  
- Analizar la evolución temporal de los movimientos para detectar posibles patrones estacionales o cambios en el comportamiento del sistema.  
- Evaluar las diferencias en los patrones de movimiento según la categoría animal, como aproximación a las distintas etapas del ciclo productivo.  
- Identificar concentraciones o dependencias dentro de la red que puedan representar puntos críticos del sistema.  


###Preguntas analíticas

Las siguientes preguntas guiarán el análisis en las etapas posteriores:

- ¿Qué regiones presentan mayor volumen de entrada y salida de ganado bovino, y cómo se distribuyen estos flujos a nivel nacional?  
- ¿Existen patrones temporales o estacionales en los movimientos bovinos a lo largo del tiempo?  
- ¿Cómo varían los patrones de movimiento según la categoría animal (por ejemplo, terneros vs. novillos), y qué implicancias tiene esto en términos del ciclo productivo?  
- ¿Se pueden identificar regiones que actúan como nodos centrales dentro de la red de movimientos bovinos?  

##3.1.B - Comprensión de los datos

In [110]:
from csv import reader, writer
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


%matplotlib inline

###***Fuente de los datos***



El dataset fue obtenido del portal de datos abiertos de Argentina (datos.gob.ar), y es publicado por SENASA.

Contiene información sobre los movimientos de ganado bovino dentro del país, incluyendo datos de origen, destino, fecha y cantidad de animales según su categoría.

Los datos están disponibles en formato CSV y para esta entrega se utilizó el archivo correspondiente al año 2018.

###***Carga de los datos***



####**Dataset_2018**

A continuación se realiza la carga del dataset utilizando la librería Pandas para su posterior análisis.

In [111]:
# Cargamos el dataset
df_2018 = pd.read_csv("movimiento-bovinos-2018.csv", encoding="latin1")


# Convertimos la columna 'fecha' a formato de fecha (datetime)
df_2018['fecha'] = pd.to_datetime(df_2018['fecha'])


# Mostramos las primeras filas del dataset para ver cómo vienen los datos
df_2018.head(10)


,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bragado,6112,0,0,0,0,11,14,0,0,0
5,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,0,0,0,35,30,0,0,0
6,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,0,0,0,0,0,0,5,0
7,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Viamonte,6385,0,0,0,0,22,28,0,30,0
8,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Hipólito Yrigoyen,6406,0,0,0,0,140,0,0,0,0
9,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Lobos,6483,50,7,0,6,10,10,0,0,0


**Visualizamos que el dataset con el que vamos a trabajar posee el siguiente orden de columnas:**
- `fecha`
- `salido_de`
- `ingresado_a`
- `provincia_origen`
- `provincia_origen_id`
- `departamento_origen`
- `departamento_origen_id`
- `provincia_destino`
- `provincia_destino_id`
- `departamento_destino`
- `departamento_destino_id`
- `vaca`
- `vaquillona`
- `novillo`
- `novillito`
- `ternero`
- `ternera`
- `torito`
- `toro`
- `bueyes`

In [112]:
# Vemos la cantidad de filas y columnas del dataset
df_2018.shape

(168802, 20)

**El dataset posee un total de 168802 filas y 20 columnas**

In [113]:
# Información general: tipos de datos, cantidad de valores y nulos
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168802 entries, 0 to 168801
Data columns (total 20 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   fecha                    168802 non-null  datetime64[ns]
 1   salido_de                168802 non-null  object        
 2   ingresado_a              168802 non-null  object        
 3   provincia_origen         168802 non-null  object        
 4   provincia_origen_id      168802 non-null  int64         
 5   departamento_origen      168802 non-null  object        
 6   departamento_origen_id   168802 non-null  int64         
 7   provincia_destino        168802 non-null  object        
 8   provincia_destino_id     168802 non-null  int64         
 9   departamento_destino     168802 non-null  object        
 10  departamento_destino_id  168802 non-null  int64         
 11  vaca                     168802 non-null  int64         
 12  vaquillona      

####**Dataset_2017**

In [114]:
# Cargamos el dataset
df_2017 = pd.read_csv("movimiento-bovinos-2017.csv", encoding="latin1")


# Convertimos la columna 'fecha' a formato de fecha (datetime)
df_2017['fecha'] = pd.to_datetime(df_2017['fecha'])


# Mostramos las primeras filas del dataset para ver cómo vienen los datos
df_2017.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,233,222,0,530,707,463,8,18,0
1,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,67,2,0,0,130,149,0,1,0
2,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,0,0,0,0,33,27,0,0,0
3,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,29,0,0,0,0
4,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,37,0,0,22,4,0,2,0
5,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,0,0,2,0,0,0,5,0
6,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Viamonte,6385,0,0,0,0,0,0,0,32,0
7,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Hipólito Yrigoyen,6406,0,0,0,0,181,119,0,0,0
8,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Las Flores,6455,0,44,0,126,0,0,0,0,0
9,2017-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Lobos,6483,140,33,0,0,22,13,0,0,0


Verificamos que posee la misma estructura de columnas que el dataset de 2018

In [115]:
# Vemos la cantidad de filas y columnas del dataset
df_2017.shape

(166622, 20)

**El dataset posee un total de 166622 filas y 20 columnas, la estructuras de columnas es igual a la del dataset 2018**

In [116]:
# Información general: tipos de datos, cantidad de valores y nulos
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168802 entries, 0 to 168801
Data columns (total 20 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   fecha                    168802 non-null  datetime64[ns]
 1   salido_de                168802 non-null  object        
 2   ingresado_a              168802 non-null  object        
 3   provincia_origen         168802 non-null  object        
 4   provincia_origen_id      168802 non-null  int64         
 5   departamento_origen      168802 non-null  object        
 6   departamento_origen_id   168802 non-null  int64         
 7   provincia_destino        168802 non-null  object        
 8   provincia_destino_id     168802 non-null  int64         
 9   departamento_destino     168802 non-null  object        
 10  departamento_destino_id  168802 non-null  int64         
 11  vaca                     168802 non-null  int64         
 12  vaquillona      

Los tipos de datos de las columnas son igual a las de 2017, por lo que podriamos unirlos para trabajarlo de forma mas integral





####**Dataset_2016**

In [117]:
# Cargamos el dataset
df_2016 = pd.read_csv("movimiento-bovinos-2016.csv", encoding="latin1")


# Convertimos la columna 'fecha' a formato de fecha (datetime)
df_2016['fecha'] = pd.to_datetime(df_2016['fecha'])


# Mostramos las primeras filas del dataset para ver cómo vienen los datos
df_2016.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,209,535,0,42,536,560,0,23,0
1,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,33,1,0,9,65,45,0,1,0
2,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,10,0,0,0,4,6,0,0,0
3,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,0,65,0,0,0
4,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bragado,6112,0,0,0,0,66,0,0,0,0
5,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,0,0,0,5,6,0,0,0
6,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Dolores,6238,0,0,0,0,0,0,0,6,0
7,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,243,0,0,11,8,0,0,0
8,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Villegas,6392,0,0,0,0,75,75,0,0,0
9,2016-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Junín,6413,0,0,0,0,20,30,0,0,0


Verificamos que posee la misma estructura de columnas que el dataset de 2018 y 2017

In [118]:
# Vemos la cantidad de filas y columnas del dataset
df_2016.shape

(163458, 20)

**El dataset posee un total de 163458 filas y 20 columnas, la estructuras de columnas es igual a la del dataset 2018 y 2017**

In [119]:
# Información general: tipos de datos, cantidad de valores y nulos
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168802 entries, 0 to 168801
Data columns (total 20 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   fecha                    168802 non-null  datetime64[ns]
 1   salido_de                168802 non-null  object        
 2   ingresado_a              168802 non-null  object        
 3   provincia_origen         168802 non-null  object        
 4   provincia_origen_id      168802 non-null  int64         
 5   departamento_origen      168802 non-null  object        
 6   departamento_origen_id   168802 non-null  int64         
 7   provincia_destino        168802 non-null  object        
 8   provincia_destino_id     168802 non-null  int64         
 9   departamento_destino     168802 non-null  object        
 10  departamento_destino_id  168802 non-null  int64         
 11  vaca                     168802 non-null  int64         
 12  vaquillona      

Los tipos de datos de las columnas son igual a las de 2018 y 2017, por lo que podriamos unirlos para trabajarlo de forma mas integral

####**Dataset_2015**

In [120]:
# Cargamos el dataset
df_2015 = pd.read_csv("movimiento-bovinos-2015.csv", encoding="latin1")


# Convertimos la columna 'fecha' a formato de fecha (datetime)
df_2015['fecha'] = pd.to_datetime(df_2015['fecha'])


# Mostramos las primeras filas del dataset para ver cómo vienen los datos
df_2015.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2015-10-01,Remate feria,Establecimiento,Ciudad Autónoma de Buenos Aires,2,Ciudad Autónoma de Buenos Aires,2000,Ciudad Autónoma de Buenos Aires,2,Ciudad Autónoma de Buenos Aires,2000,11,0,0,0,0,0,0,0,0
1,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,285,302,0,160,828,1739,0,84,0
2,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,15,126,0,67,47,54,0,0,0
3,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Adolfo Gonzales Chaves,6014,0,0,0,0,0,0,0,3,0
4,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,94,35,0,0,37,40,0,0,0
5,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Ayacucho,6042,0,50,0,0,0,0,0,3,0
6,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Azul,6049,0,0,0,0,0,0,0,3,0
7,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Benito Juárez,6084,0,0,0,0,0,0,0,3,0
8,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,11,0,0,0,1,0,0,0,0
9,2015-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chascomús,6218,100,0,0,0,37,0,0,0,0


Verificamos que posee la misma estructura de columnas que el dataset de 2018, 2017 y 2016

In [121]:
# Vemos la cantidad de filas y columnas del dataset
df_2015.shape

(163483, 20)

**El dataset posee un total de 163483 filas y 20 columnas, la estructuras de columnas es igual a la del dataset 2018, 2017 y 2016**

In [122]:
# Información general: tipos de datos, cantidad de valores y nulos
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168802 entries, 0 to 168801
Data columns (total 20 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   fecha                    168802 non-null  datetime64[ns]
 1   salido_de                168802 non-null  object        
 2   ingresado_a              168802 non-null  object        
 3   provincia_origen         168802 non-null  object        
 4   provincia_origen_id      168802 non-null  int64         
 5   departamento_origen      168802 non-null  object        
 6   departamento_origen_id   168802 non-null  int64         
 7   provincia_destino        168802 non-null  object        
 8   provincia_destino_id     168802 non-null  int64         
 9   departamento_destino     168802 non-null  object        
 10  departamento_destino_id  168802 non-null  int64         
 11  vaca                     168802 non-null  int64         
 12  vaquillona      

Los tipos de datos de las columnas son igual a las de los datasets de 2018, 2017 y 2016, por lo que podriamos unirlos para trabajarlo de forma mas integral

####**Unimos a todos los datasets en uno para poder trabajarlo de mejor forma**

In [123]:
#realizamos un condicional previo para ver que no existan diferencias entre las columnas, en caso de no existir, dejamos que se unan
if (df_2018.columns.equals(df_2017.columns) and df_2018.columns.equals(df_2016.columns) and df_2018.columns.equals(df_2015.columns)):
  try:
    df_unido = pd.concat([df_2018,df_2017,df_2016,df_2015],ignore_index=True)
  except Exception as e:
    print(f"Error al unir los datasets: {e}")
else:
  print("Los datasets no poseen las mismas columnas")

###**Exploramos los tipos de datos de las 20 para luego armar el diccionario**

In [124]:
# Contamos la cantidad de valores únicos en cada columna
df_unido.nunique()

,0
fecha,48
salido_de,3
ingresado_a,3
provincia_origen,24
provincia_origen_id,24
departamento_origen,406
departamento_origen_id,480
provincia_destino,24
provincia_destino_id,24
departamento_destino,413


**Verificamos a partir de esto que contamos con:**
- medicion correspondiente a 49 meses (12 meses * 4 años de mediciones)
- que hay 3 tipos de establecimientos de salida y de origen
- que los registros comprenden las 24 provincias pero un total de 406 departamentos de origen y 489 departamentos de destino


In [125]:
departamentos_descripcion_unicos = df_unido.groupby('departamento_destino')['departamento_destino_id'].nunique()
departamentos_descripcion_unicos

,departamento_destino_id
departamento_destino,
12 de Octubre,1
1° de Mayo,1
2 de Abril,1
25 de Mayo,5
9 de Julio,4
...,...
Yerba Buena,1
Zapala,1
Zárate,1


Podemos ver que hay departamentos que poseen la misma descripcion pero lo podemos encontrar con numeros de id diferentes, lo que teniendo en cuenta el pais al que pertenecen pueden ser descripciones de departamentos que se llaman igual en diferentes provincias, por lo cual la celda descripcion no identifica de forma univoca

In [126]:
departamentos_id_unicos = df_unido.groupby('departamento_destino_id')['departamento_destino'].nunique()
departamentos_id_unicos

,departamento_destino
departamento_destino_id,
2000,1
6007,1
6014,1
6021,1
6028,1
...,...
90105,1
90112,1
90119,1


Terminamos de confirmar que lo que identifica univocamente es el id de departamento ya que muestra una relacion 1-1

In [127]:
# Obtenemos estadísticas básicas de las variables numéricas
df_unido.describe()

,fecha,provincia_origen_id,departamento_origen_id,provincia_destino_id,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
count,662365,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000,662365.000000
mean,2016-12-24 01:09:57.482657280,27.464812,27677.970768,29.417403,29650.950920,39.441153,33.162644,26.509032,37.058877,40.664934,30.051515,0.859481,3.232853,0.001285
min,2015-01-01 00:00:00,2.000000,2000.000000,2.000000,2000.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,2016-01-01 00:00:00,6.000000,6511.000000,6.000000,6532.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,2017-01-01 00:00:00,14.000000,14182.000000,14.000000,14105.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,2018-01-01 00:00:00,42.000000,42056.000000,46.000000,46084.000000,30.000000,25.000000,10.000000,26.000000,32.000000,20.000000,0.000000,1.000000,0.000000
max,2018-12-01 00:00:00,94.000000,94015.000000,94.000000,94015.000000,10027.000000,8242.000000,7856.000000,8264.000000,8867.000000,6970.000000,1075.000000,1541.000000,262.000000
std,NaN,26.849974,26737.810207,29.326809,29196.516691,148.395899,120.316485,114.919507,143.184826,139.273146,112.602110,8.875347,16.345925,0.339246


Con el metodo describe() podemos ver que dentro el dataset refleja una distribucion de ganado demasiado extrema debido a que las medias identificadas son demasiado altas, las medianas estan bajas y hay mucha presencia de 0 lo que indica que la distribucion esta extremadamente distribuida y que encontraremos outliers grandes

In [128]:
# Listamos los nombres de todas las columnas del dataset
df_unido.columns

Index(['fecha', 'salido_de', 'ingresado_a', 'provincia_origen',
       'provincia_origen_id', 'departamento_origen', 'departamento_origen_id',
       'provincia_destino', 'provincia_destino_id', 'departamento_destino',
       'departamento_destino_id', 'vaca', 'vaquillona', 'novillo', 'novillito',
       'ternero', 'ternera', 'torito', 'toro', 'bueyes'],
      dtype='object')

In [129]:
df_unido.shape

(662365, 20)

Vemos que la sumatoria de los datasets nos entregan un dataset de 20 columnas con un total de 662365 filas

###***Estructura del dataset***



El dataset contiene una gran cantidad de registros (más de 662.000 filas) y aproximadamente 20 columnas.

Cada fila representa un movimiento de ganado entre un origen y un destino específico.

Se pueden identificar distintos tipos de variables:

- Variables temporales: como la fecha
- Variables categóricas: provincias, departamentos, tipo de establecimiento
- Variables numéricas: cantidad de animales según categoría (vacas, novillos, terneros, etc.)

Esto permite analizar tanto la distribución geográfica como la composición del ganado en movimiento.


In [130]:
# Contamos la cantidad de valores nulos por cada columna
df_unido.isnull().sum()

,0
fecha,0
salido_de,0
ingresado_a,0
provincia_origen,0
provincia_origen_id,0
departamento_origen,0
departamento_origen_id,0
provincia_destino,0
provincia_destino_id,0
departamento_destino,0


Contabilizamos la cantidad de nulos por columna a fin de identificarlos

In [131]:
# Porcentaje de valores nulos por columna
(df_unido.isnull().sum() / len(df_unido)) * 100

,0
fecha,0.0
salido_de,0.0
ingresado_a,0.0
provincia_origen,0.0
provincia_origen_id,0.0
departamento_origen,0.0
departamento_origen_id,0.0
provincia_destino,0.0
provincia_destino_id,0.0
departamento_destino,0.0


A partir del porcentaje terminamos de confirmar y asegurar que no poseemos ningun tipo de dato nulo por columna

###***Calidad de los datos***



Se realizó un análisis preliminar de calidad de los datos, observando que no se presentan valores nulos en ninguna de las variables.

Esto indica que el dataset se encuentra completo desde el punto de vista estructural, lo cual facilita su análisis.

Sin embargo, esto no descarta la posible existencia de otras inconsistencias, como errores en categorías o valores atípicos, que deberán analizarse en etapas posteriores.

####***A partir de los datos explorados del dataset logramos crear el siguiente diccionario***
| Nombre de la columna      | Descripción                                        | Tipo de dato | Valores nulos | Posible uso en análisis/modelado              |
| ------------------------- | -------------------------------------------------- | ------------ | ------------- | --------------------------------------------- |
| `fecha`                   | Fecha en que se realiza el movimiento              | `datetime64` | Ninguno       | Análisis temporal, estacionalidad, tendencias |
| `salido_de`               | Lugar desde el que sale el ganado                  | `object`     | Ninguno       | Análisis logístico y de trazabilidad          |
| `ingresado_a`             | Lugar al que llega el ganado                       | `object`     | Ninguno       | Identificación de destinos frecuentes         |
| `provincia_origen`        | Nombre de la provincia de origen                   | `object`     | Ninguno       | Distribución geográfica                       |
| `provincia_origen_id`     | ID de la provincia de origen                       | `int64`      | Ninguno       | Codificación para modelos                     |
| `departamento_origen`     | Nombre del departamento de origen                  | `object`     | Ninguno       | Segmentación regional                         |
| `departamento_origen_id`  | ID del departamento de origen                      | `int64`      | Ninguno       | Variables categóricas para ML                 |
| `provincia_destino`       | Nombre de la provincia de destino                  | `object`     | Ninguno       | Flujo interprovincial                         |
| `provincia_destino_id`    | ID de la provincia de destino                      | `int64`      | Ninguno       | Modelos predictivos y agrupaciones            |
| `departamento_destino`    | Nombre del departamento de destino                 | `object`     | Ninguno       | Análisis territorial                          |
| `departamento_destino_id` | ID del departamento de destino                     | `int64`      | Ninguno       | Relaciones origen-destino                     |
| `vaca`                    | Hembras que han parido al menos una vez            | `int64`      | Ninguno       | Producción reproductiva                       |
| `vaquillona`              | Hembras jóvenes que no han parido                  | `int64`      | Ninguno       | Proyección de stock reproductivo              |
| `novillo`                 | Machos castrados adultos                           | `int64`      | Ninguno       | Producción cárnica                            |
| `novillito`               | Machos castrados jóvenes                           | `int64`      | Ninguno       | Evolución de categorías bovinas               |
| `ternero`                 | Machos menores de un año                           | `int64`      | Ninguno       | Indicadores de nacimientos                    |
| `ternera`                 | Hembras menores de un año                          | `int64`      | Ninguno       | Crecimiento del rodeo                         |
| `torito`                  | Machos jóvenes mayores de un año sin servicio      | `int64`      | Ninguno       | Potencial reproductivo                        |
| `toro`                    | Machos destinados a reproducción                   | `int64`      | Ninguno       | Capacidad reproductiva                        |
| `bueyes`                  | Machos adultos castrados para trabajo o infértiles | `int64`      | Ninguno       | Uso productivo tradicional                    |





##3.1.C Justificación del dataset



Los dataset seleccionados resultan adecuados para este trabajo porque permiten analizar de forma concreta los movimientos de ganado bovino en Argentina, que es justamente el foco del problema planteado.

Se trata de un conjunto de datos amplio, con una gran cantidad de registros y variables, lo que posibilita realizar un análisis significativo y obtener conclusiones relevantes. Además, incluye tanto variables numéricas (cantidad de animales por categoría) como categóricas (provincias, departamentos, tipo de establecimiento), lo que permite abordar el análisis desde distintas perspectivas.

Otro aspecto importante es que incorpora una dimensión temporal, lo cual hace posible estudiar la evolución de los movimientos y detectar posibles patrones a lo largo del tiempo, algo directamente vinculado con las preguntas analíticas definidas.

Finalmente, existe una relación directa entre los datos disponibles y los objetivos del trabajo, ya que el dataset permite analizar los flujos de ganado entre regiones, comparar comportamientos según categorías animales e identificar posibles patrones o concentraciones dentro del sistema.

En este sentido, los datasets no solo cumplen con los requisitos técnicos de la consigna, sino que también resulta pertinente desde el punto de vista del problema planteado y las preguntas analíticas que guían el proyecto.

##4.1.A Preparacion de los datos

###Correccion, limpieza y formateo de los datos

En el dataset no se encuentran datos faltantes, ni tampoco nulos

In [132]:
df_unido.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bragado,6112,0,0,0,0,11,14,0,0,0
5,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,0,0,0,35,30,0,0,0
6,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,0,0,0,0,0,0,5,0
7,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Viamonte,6385,0,0,0,0,22,28,0,30,0
8,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Hipólito Yrigoyen,6406,0,0,0,0,140,0,0,0,0
9,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Lobos,6483,50,7,0,6,10,10,0,0,0


In [133]:
columnas_upper = [
    "salido_de",
    "ingresado_a",
    "provincia_origen",
    "departamento_origen",
    "provincia_destino",
    "departamento_destino"
]

for col in columnas_upper:
    if col in df_unido.columns:
        df_unido[col] = df_unido[col].astype(str).str.upper()
    else:
        print(f"No existe la columna: {col}")

df_unido

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLÍVAR,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,6112,0,0,0,0,11,14,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORÍFICO,SANTA FE,82,VERA,82133,ENTRE RÍOS,30,PARANÁ,30084,45,0,0,0,0,0,0,0,0
662361,2015-12-01,REMATE FERIA,FRIGORÍFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,82077,97,1,0,1,0,0,0,0,0
662362,2015-12-01,REMATE FERIA,FRIGORÍFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTÓBAL,82091,32,1,9,4,0,0,0,4,0
662363,2015-12-01,REMATE FERIA,FRIGORÍFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,82133,43,1,6,2,0,0,0,1,0


Normalizamos todas las celdas de texto a mayusculas para mejorar la manipulacion de los datos

In [134]:
#A continuacion realizaremos una limpieza en las celdas de texto, quitamos acentos, espacios en blanco al comienzo o final y posibilidad de dobles espacios
columnas_ascii = [
    "salido_de",
    "ingresado_a",
    "provincia_origen",
    "departamento_origen",
    "provincia_destino",
    "departamento_destino"
]
for col in columnas_ascii:
    if col in df_unido.columns:
      df_unido[col] = (
          df_unido[col]
          .astype(str)
          .str.normalize('NFKD')
          .str.encode('ascii', errors='ignore')
          .str.decode('utf-8')
          .str.strip()
          .str.replace(r'\s+', ' ', regex=True))
    else:
        print(f"No existe la columna: {col}")

df_unido

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLIVAR,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,6112,0,0,0,0,11,14,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,ENTRE RIOS,30,PARANA,30084,45,0,0,0,0,0,0,0,0
662361,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,82077,97,1,0,1,0,0,0,0,0
662362,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTOBAL,82091,32,1,9,4,0,0,0,4,0
662363,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,82133,43,1,6,2,0,0,0,1,0


Terminamos de limpiar las celdas de texto quitando posibles acentos, espacios en blanco al comienzo o al final, y tambien eliminamos los posibles doble espacios intermedios que puedan existir

In [135]:
# buscamos fechas en formato extraño y las convertimos
# a un formato más entendible para el cliente

df_unido['fecha'] = pd.to_datetime(
    df_unido['fecha'],
    errors='coerce'
)

# mostramos fechas inválidas
fechas_invalidas = df_unido[df_unido['fecha'].isna()]
if not fechas_invalidas.empty:
    print("Fechas inválidas encontradas:")
    print(fechas_invalidas)
else:
    print("No se encontraron fechas inválidas.")


df_unido

No se encontraron fechas inválidas.


,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLIVAR,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,6112,0,0,0,0,11,14,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,ENTRE RIOS,30,PARANA,30084,45,0,0,0,0,0,0,0,0
662361,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,82077,97,1,0,1,0,0,0,0,0
662362,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTOBAL,82091,32,1,9,4,0,0,0,4,0
662363,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,82133,43,1,6,2,0,0,0,1,0


Verificamos que el dataset no posee formatos de fechas invalidos

In [136]:
def obtener_estacion(fecha):
  mes = fecha.month
  if mes in [12,1,2]:
    return 'VERANO'
  elif mes in [3,4,5]:
    return 'OTONIO'
  elif mes in [6,7,8]:
    return 'INVIERNO'
  elif mes in [9,10,11]:
    return 'PRIMAVERA'

df_unido['estacion'] = df_unido['fecha'].apply(obtener_estacion)
df_unido

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,...,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes,estacion
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,...,212,150,0,342,931,566,0,21,0,VERANO
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,...,35,0,0,0,96,104,0,4,0,VERANO
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,...,0,0,0,0,31,9,0,0,0,VERANO
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLIVAR,...,0,0,0,0,29,39,0,0,0,VERANO
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,...,0,0,0,0,11,14,0,0,0,VERANO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,ENTRE RIOS,30,PARANA,...,45,0,0,0,0,0,0,0,0,VERANO
662361,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,...,97,1,0,1,0,0,0,0,0,VERANO
662362,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTOBAL,...,32,1,9,4,0,0,0,4,0,VERANO
662363,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,...,43,1,6,2,0,0,0,1,0,VERANO


#####Creamos nuevas columnas que nos ayudaran a mejorar el analisis posterior

Creamos una nueva columna en donde segun el mes del anio le asignamos una estacion para poder agrupar luego los movimientos segun estaciones

In [137]:
rows=[
    'vaca',
    'vaquillona',
    'novillo',
    'novillito',
    'ternero',
    'ternera',
    'torito',
    'toro',
    'bueyes'
]

df_unido['total_movimientos'] = df_unido[rows].sum(axis=1)
df_unido

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,...,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes,estacion,total_movimientos
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,...,150,0,342,931,566,0,21,0,VERANO,2222
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,...,0,0,0,96,104,0,4,0,VERANO,239
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,...,0,0,0,31,9,0,0,0,VERANO,40
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLIVAR,...,0,0,0,29,39,0,0,0,VERANO,68
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,...,0,0,0,11,14,0,0,0,VERANO,25
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,ENTRE RIOS,30,PARANA,...,0,0,0,0,0,0,0,0,VERANO,45
662361,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,...,1,0,1,0,0,0,0,0,VERANO,99
662362,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTOBAL,...,1,9,4,0,0,0,4,0,VERANO,50
662363,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,...,1,6,2,0,0,0,1,0,VERANO,53


Creamos una nueva columna en donde sumamos el total de los animales que fueron transportados en un registro (fila)

In [138]:
df_unido['anio'] = df_unido['fecha'].dt.year
df_unido['mes'] = df_unido['fecha'].dt.month

Creamos dos columnas nuevas, anio y mes para desglosar la fecha y poder mejorar la manipulacion a la hora de realizar analisis temporales o crear variables nuevas

#####Guardamos y volvemos a cargar el dataset para evitar errores de lectura

In [139]:
df_unido.to_csv(
    'df_unido_limpio.csv',
    index=False,
    encoding='utf-8'
)

Guardamos el dataframe limpio con las columnas que hemos creado

In [140]:
#volvemos a leer el dataset limpio
df_limpio = pd.read_csv('df_unido_limpio.csv', parse_dates=['fecha'])
df_limpio


,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,...,novillito,ternero,ternera,torito,toro,bueyes,estacion,total_movimientos,anio,mes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,...,342,931,566,0,21,0,VERANO,2222,2018,1
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,...,0,96,104,0,4,0,VERANO,239,2018,1
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,...,0,31,9,0,0,0,VERANO,40,2018,1
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLIVAR,...,0,29,39,0,0,0,VERANO,68,2018,1
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,...,0,11,14,0,0,0,VERANO,25,2018,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
662360,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,ENTRE RIOS,30,PARANA,...,0,0,0,0,0,0,VERANO,45,2015,12
662361,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,9 DE JULIO,...,1,0,0,0,0,0,VERANO,99,2015,12
662362,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,SAN CRISTOBAL,...,4,0,0,0,4,0,VERANO,50,2015,12
662363,2015-12-01,REMATE FERIA,FRIGORIFICO,SANTA FE,82,VERA,82133,SANTA FE,82,VERA,...,2,0,0,0,1,0,VERANO,53,2015,12


volvemos a leer el dataset limpio para asegurarnos de que trabajaremos con el correcto

#####Variables

In [142]:
#movimientos de cada categoria por anio
categorias = [
    'vaca',
    'vaquillona',
    'novillo',
    'novillito',
    'ternero',
    'ternera',
    'torito',
    'toro',
    'bueyes'
]

movimientos_por_anio = (
    df_limpio
    .groupby('anio')[categorias]
    .sum()
)

print(movimientos_por_anio)

         vaca  vaquillona  novillo  novillito  ternero  ternera  torito  \
anio                                                                      
2015  6084473     5295884  4410078    6060589  6259728  4456380   97641   
2016  6258298     4957152  4296674    5972040  6609621  4648047  106634   
2017  6626432     5667294  4405498    6200185  6871758  5180413  148607   
2018  7155236     6045445  4446405    6313689  7193922  5620232  216408   

        toro  bueyes  
anio                  
2015  537349     104  
2016  514200     121  
2017  530424     132  
2018  559356     494  


Creamos la variable movimientos_por_anio para contabilizar los movimientos producidos en un anio para cada tipo de animal

In [144]:
movimientos_por_estacion_por_anio = (
   df_limpio
   .groupby(['anio', 'estacion'])['total_movimientos']
   .sum()
)

print(movimientos_por_estacion_por_anio)

anio  estacion 
2015  INVIERNO      8466444
      OTONIO        9647613
      PRIMAVERA     7667818
      VERANO        7420351
2016  INVIERNO      8506979
      OTONIO        9701369
      PRIMAVERA     7485340
      VERANO        7669099
2017  INVIERNO      9469035
      OTONIO       10138606
      PRIMAVERA     8324782
      VERANO        7698320
2018  INVIERNO      9840207
      OTONIO       11027226
      PRIMAVERA     8254889
      VERANO        8428865
Name: total_movimientos, dtype: int64


Creamos la variable movimientos_por_estacion_por_anio para poder consutlar el total de los movimientos producidos en cada estacion del anio

In [143]:
#movimientos por categoria segun estacion del anio\
categorias = [
    'vaca',
    'vaquillona',
    'novillo',
    'novillito',
    'ternero',
    'ternera',
    'torito',
    'toro',
    'bueyes'
    ]

movimientos_por_estacion_por_anio_segun_categoria = (
   df_limpio
   .groupby(['anio', 'estacion'])[categorias]
   .sum()
)

print(movimientos_por_estacion_por_anio_segun_categoria)


                   vaca  vaquillona  novillo  novillito  ternero  ternera  \
anio estacion                                                               
2015 INVIERNO   1733647     1188555  1155628    1377814  1667196  1186642   
     OTONIO     1635902     1345903  1208984    1546434  2292895  1492048   
     PRIMAVERA  1402504     1334785   998729    1518022  1211741   987933   
     VERANO     1312420     1426641  1046737    1618319  1087896   789757   
2016 INVIERNO   1846673     1104035  1064816    1359630  1722084  1249485   
     OTONIO     1689587     1186372  1175314    1470808  2502160  1560513   
     PRIMAVERA  1392008     1306524   945697    1478869  1172173   979447   
     VERANO     1330030     1360221  1110847    1662733  1213204   858602   
2017 INVIERNO   2013973     1319077  1157390    1432642  1906112  1466389   
     OTONIO     1768263     1365827  1205238    1567530  2454235  1650925   
     PRIMAVERA  1513513     1518482  1049729    1575140  1294520  1134863   

Creamos la variable movimientos_por_estacion_por_anio para contabilizar los movimientos producidos en una estacion del anio para cada tipo de animal

##4.1.B Análisis descriptivo


- Calcular e interpreinterpretar medidas de tendencia central y dispersión para las variables
numéricas relevantes.
- Construir tablas de frecuencia y proporciones para las variables categóricas.
- Identificar y analizar valores atípicos: describir su impacto potencial en el análisis.

##4.1.C Análisis de relaciones entre variables

- Explorar correlaciones y asociaciones entre variables.
- Conectar los hallazgos con las preguntas analíticas formuladas en la Entrega 1.

##4.1.D Visualizaciones

- Producir al menos 5 (cinco) visualizaciones distintas que aporten información
sustantiva al análisis.
- Cada visualización debe incluir título, etiquetas claras en los ejes y una interpretación
escrita en celda Markdown.
- Aplicar principios de diseño: elegir el tipo de gráfico adecuado para cada variable o
relación, evitar gráficos decorativos o redundantes.

##4.1.E Sintesis del EDA

Escribir un apartado de síntesis que reponda:
- ¿Cuáles son los principales hallazgos del análisis exploratorio?
- ¿Qué patrones o tendencias se observan?
- ¿Qué preguntas de la Entrega 1 pueden responderse ya con el EDA, y cuales
requieren un abordaje más avanzado?
- ¿Hay indicios de que un modelo predictivo podría ser útil? ¿Por qué?

##7. Conclusión



En esta primera etapa del trabajo se logró definir el problema, entender el contexto del dominio y realizar una exploración inicial del dataset.

Se identificaron las principales características de los datos y algunos aspectos a mejorar en términos de calidad.

Este análisis inicial sirve como base para las siguientes etapas, donde se profundizará en el análisis exploratorio y se buscarán patrones más concretos en los datos.